### PR TITLE
Add support for RSA-SHA1 signatures

### LINKED ISSUE #30: Lack of RSA-SHA1 support
HI guys!

Have a trouble with authenticating w/ JIRA REST API, because they support only RSA-SHA1(not SHA2).
Will there be support in the future of RSA-SHA1 ?

### PR DESCRIPTION
**Pull request by erickoledadevrel**
Now that RSA-SHA1 signing is possible in Apps Script, add support to the library. Add a sample for the Xero API which uses RSA-SHA1. Also clarify in the readme the support for the different OAuth flows supported.

Fixes #30.

### CODE REVIEW

File: src/Signer.gs [hunk 1]
```diff
@@ -170,6 +174,13 @@
   OAuth.prototype.getSigningKey = function(token_secret) {
     token_secret = token_secret || '';
 
+    // Don't percent encode the signing key (PKCS#8 PEM private key) when using
+    // the RSA-SHA1 method. The token secret is never used with the RSA-SHA1
+    // method.
+    if (this.signature_method == 'RSA-SHA1') {
```
Comment (grant) [inline src/Signer.gs]: Use `===`.
https://gist.github.com/nuno/8378948#conditional-expressions--equality

File: samples/XeroPrivate.gs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  var result = JSON.parse(response.getContentText());
+  Logger.log(JSON.stringify(result, null, 2));
+}
+
+/**
+ * Reset the authorization state, so that it can be re-tested.
+ */
+function reset() {
+  var service = getService();
+  service.reset();
```
Comment (grant) [inline samples/XeroPrivate.gs]: Wouldn't this be easier?
```js
getService().reset();
```

File: samples/XeroPrivate.gs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+/**
+ * This sample that demonstrates how to configured the library for the Xero API,
+ * when using a private application. Although the Xero documentation calls it a
+ * "2 legged" flow it is really a 1-legged flow. Public Xero applications use
+ * a 3-legged flow not shown here.
+ * https://developer.xero.com/documentation/auth-and-limits/private-applications
```
Comment (grant) [inline samples/XeroPrivate.gs]: Maybe use [JSDoc links](http://usejsdoc.org/tags-inline-link.html)? `{@link URL}`

File: src/Signer.gs [hunk 4]
```diff
@@ -170,6 +174,13 @@
   OAuth.prototype.getSigningKey = function(token_secret) {
     token_secret = token_secret || '';
 
+    // Don't percent encode the signing key (PKCS#8 PEM private key) when using
```
Comment (grant) [inline src/Signer.gs]: This comment could be 2 lines instead of 3.

File: src/Signer.gs (hunk 1 above)
Comment (erickoledadevrel) [reply src/Signer.gs]: Done

File: samples/XeroPrivate.gs (hunk 2 above)
Comment (erickoledadevrel) [reply samples/XeroPrivate.gs]: Done

File: samples/XeroPrivate.gs (hunk 3 above)
Comment (erickoledadevrel) [reply samples/XeroPrivate.gs]: Done

File: src/Signer.gs (hunk 4 above)
Comment (erickoledadevrel) [reply src/Signer.gs]: Not if I want to stay within 80 chars. This file doesn't stay within 80 chars (it's 3rd party) but the rest of the library does.

**Pull request merged**